app/services/workflow_executions/preparation_service.rb

```diff
@@ -52,10 +57,18 @@ def execute # rubocop:disable Metrics/MethodLength
 
     private
 
-    def parse_samplesheet_headers
-      workflow = Irida::Pipelines.instance.find_pipeline_by(@workflow_execution.metadata['workflow_name'],
-                                                            @workflow_execution.metadata['workflow_version'])
-      workflow.samplesheet_headers
+    def find_pipeline
+      Irida::Pipelines.instance.find_pipeline_by(@workflow_execution.metadata['workflow_name'],
+                                                 @workflow_execution.metadata['workflow_version'])
+    end
+
+    def validate_pipeline
```

Comment:
We probably want to update the job to also set the state to error as the pipeline can be marked as non-executable after submission and before the preparation job actually runs